### DISCUSSION TITLE
Exporting Rules ?

### DISCUSSION THREAD

**Discussion by R-Nabil** [firefly-iii/firefly-iii#9099] [General]
Hi,

Is there a way to export rules ? I got an old instance of FF3 that I'm replacing with a cleaner ones, but i'd like to keep the rules I have from the old ones. Is there a way I can export them or copy them across ?

Thanks

**Comment by JC5** [comment]
Mmm, you could use the API, but it needs some technical finesse, and perhaps a tool like Postman. You could request the rules over the API from one side, and  submit thm on the other side.

**Reply by R-Nabil** [reply]
I suppose there is no hint or example somewhere ? I do have technical knowledge but not sure if that's enough here :)

**Reply by R-Nabil** [reply]
Ok found it : https://api-docs.firefly-iii.org/#/data/exportRules

**Reply by JC5** [reply]
Sorry, I didn't have the time yet to answer. Exporting as CSV works, but importing is tricky. It's easier to simply list the rules using `GET /api/v1/rules` because you can copy/paste it (one rule at a time) in `POST /api/v1/rules`

**Reply by R-Nabil** [reply]
> Sorry, I didn't have the time yet to answer. Exporting as CSV works, but importing is tricky. It's easier to simply list the rules using `GET /api/v1/rules` because you can copy/paste it (one rule at a time) in `POST /api/v1/rules`

Thank you of course no problem. 

I have another question : when importing my transactions history it takes a huge amount of time through the data importer. It would be ok but many times it crashed. 

Would it make sense to do it through API or would it not necessarily be any faster ?

**Reply by JC5** [reply]
> Would it make sense to do it through API or would it not necessarily be any faster ?

Try using the command line, that will not stop working at least. If you have a lot of rules, it will be slow unfortunately.

**Reply by R-Nabil** [reply]
I figured out one of the main reason for the slowness was that my pgsql was limited in resources. I increased the limit and it's still slow but better and manageable. 

Thank you again for your help
